### PR TITLE
move power series to Parent

### PR DESCRIPTION
**Pull request by fchapoton**
make the power series rings inherit from Parent instead of the auld `Ring` class

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

**Comment by fchapoton** [comment]
the case of Laurent series triggers a failure in Drinfeld modules, so it is kept for another pull request

File: src/sage/rings/multi_power_series_ring.py [hunk 1]
```diff
@@ -1019,6 +1020,19 @@ def ngens(self):
...
+        """
+        Return the generators of this ring.
+
+        EXAMPLES::
+
+            sage: M = PowerSeriesRing(ZZ, 3, 'v')
+            sage: M.gens()
+            (v0, v1, v2)
+        """
+
```
Comment (mantepse) [inline src/sage/rings/multi_power_series_ring.py]: This blank line is unconventional, no?

Comment (fchapoton) [reply src/sage/rings/multi_power_series_ring.py]: oui, c'est vrai, merci. J'ai corrigé ça.

**Review by mantepse — APPROVED** [review]
merci !

**Pull request merged**